pkg/console/operator/operator.go

```diff
@@ -142,18 +142,34 @@ func NewConsoleOperator(
 	configNameFilter := util.IncludeNamesFilter(api.ConfigResourceName)
 	targetNameFilter := util.IncludeNamesFilter(api.OpenShiftConsoleName)
 
-	olmConfigInformer := dynamicInformers.ForResource(schema.GroupVersionResource{Group: api.OLMConfigGroup, Version: api.OLMConfigVersion, Resource: api.OLMConfigResource})
+	informers := []factory.Informer{
+		configV1Informers.Consoles().Informer(),
+		operatorConfigInformer.Informer(),
+		configV1Informers.Infrastructures().Informer(),
+		configV1Informers.Ingresses().Informer(),
+		configV1Informers.Proxies().Informer(),
+		configV1Informers.OAuths().Informer(),
+	}
+
+	olmGroupVersionResource := schema.GroupVersionResource{
+		Group:    api.OLMConfigGroup,
+		Version:  api.OLMConfigVersion,
+		Resource: api.OLMConfigResource,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+	defer cancel()
+
+	// We check if the resource exists in the cluster at all before attaching an informer
+	if _, err := dynamicClient.Resource(olmGroupVersionResource).List(ctx, metav1.ListOptions{}); !isNoMatchError(err) {
```

Comment:
@eggfoobar what if the OLM capability is enabled post installation? Can the capability be enable at any time? or is there any restriction ?
In case the capability will be disabled at start and then enabled, I dont think the informer will be registered, since we are registering the informers in the `NewConsoleOperator` constructor.
Wondering if it wont be better to restart the operator's pod on this change?

Reply:
Yeah, that might be a good idea, I added some WIP logic to get this running in my e2e tests that just checks if the resource exists but in the future it would be preferable to have an informer on the clusterversions and if we have the resource enabled, we trigger a reboot then. There might be a more elegant way of doing that, but that's what came to mind, what do you think?

Reply:
So this is something I had in mind, to basically kill the pod end force new rollout so the informer is registered 👍